client/js/gdpr/table.js

```diff
@@ -0,0 +1,80 @@
+export class GdprTable {
+  /**
+   * @param {HTMLDivElement} chart The element that contains the chart.
+   * @param {Map<string, number[]>} histories The ranking history for each
+   * track.
+   * @param {Date[]} dates The date of each history entry.
+   */
+  constructor(chart, histories, dates) {
+    this.table = document.createElement('table');
+    this.chart = chart;
+    this.histories = histories;
+    this.dates = dates;
+    this.setup();
+  }
+
+  /**
+   * Sets up the GDPR table by (attaching event handlers, etc.)
+   *
+   * @private
+   */
+  setup() {
+    this.chart.addEventListener('series-hit', this.onSeriesHit.bind(this));
+  }
+
+  /**
+   * Fires when the user hovers over a series on the chart.
+   *
+   * @param {CustomEvent} ev The 'series-hit' event object.
+   * @private
+   */
+  onSeriesHit(ev) {
+    const {x, y} = ev.detail;
+    const ranking = [];
+
+    for (const [key, history] of this.histories) {
+      // song's rank at this x position (i.e., date)
+      const rank = history[x];
+      // true if this is the song the user hovered over
+      const selected = rank === y;
+
+      if (rank === undefined || rank === null) {
+        continue;
+      }
+
+      ranking.push({key, rank, selected});
+    }
+
+    // sort by rank ascending
+    ranking.sort((a, b) => a.rank < b.rank ? -1 : 1);
```

Comment:
Is `rank` stored as a number? If so, why not `ranking.sort((a,b) => a.rank - b.rank)`?

Reply:
That would work too, it just didn't occur to me. It seems to have the same behaviour, but your version might be easier to read.